deepchecks/vision/vision_data.py

```diff
@@ -0,0 +1,353 @@
+                 num_classes: Optional[int] = None,
+                 label_map: Optional[Dict[int, str]] = None,
+                 transform_field: Optional[str] = 'transforms'):
+
+        self._data_loader = self._get_data_loader_copy(data_loader)
+
+        self._num_classes = num_classes
+        self._label_map = label_map
+        self._transform_field = transform_field
+        self._warned_labels = set()
+        self._has_images = False
+
+        try:
+            self.validate_image_data(next(iter(self._data_loader)))
+            self._has_images = True
+        except DeepchecksNotImplementedError:
+            logger.warning('batch_to_images() was not implemented, some checks will not run')
+        except ValidationError as ex:
+            logger.warning('batch_to_images() was not implemented correctly, '
+                           'the validiation has failed with the error: %s', {str(ex)})
+
+        self._n_of_samples_per_class = None
+        self._task_type = None
+        self._has_label = None
+        self._current_seed = None
+
+    @abstractmethod
+    def get_classes(self, batch_labels: Union[List[torch.Tensor], torch.Tensor]):
+        """Get a labels batch and return classes inside it."""
+        return NotImplementedError('get_classes() must be implemented in a subclass')
+
+    @abstractmethod
+    def batch_to_labels(self, batch) -> Union[List[torch.Tensor], torch.Tensor]:
+        """Transform a batch of data to labels."""
+        raise DeepchecksNotImplementedError('batch_to_labels() must be implemented in a subclass')
+
+    @abstractmethod
+    def infer_on_batch(self, batch, model, device) -> Union[List[torch.Tensor], torch.Tensor]:
+        """Infer on a batch of data."""
+        raise DeepchecksNotImplementedError('infer_on_batch() must be implemented in a subclass')
+
+    @abstractmethod
+    def validate_label(self, batch):
+        """Validate a batch of labels."""
+        raise NotImplementedError('validate_label() must be implemented in a subclass')
+
+    @abstractmethod
+    def validate_prediction(self, batch, model, device):
+        """Validate a batch of predictions."""
+        raise DeepchecksValueError(
+            'validate_prediction() must be implemented in a subclass'
+        )
+
+    @abstractmethod
+    def batch_to_images(self, batch) -> Iterable[np.ndarray]:
+        """
+        Transform a batch of data to images in the accpeted format.
+
+        Parameters
+        ----------
+        batch : torch.Tensor
+            Batch of data to transform to images.
+
+        Returns
+        -------
+        Iterable[np.ndarray]
+            List of images in the accepted format. Each image in the iterable must be a [H, W, C] 3D numpy array.
+            See notes for more details.
+            :func: `batch_to_images` must be implemented in a subclass.
+
+        Examples
+        --------
+        >>> def batch_to_images(self, batch):
+        ...     return batch[0]
```

Comment:
I think the example we need here is the one that applies to imagenet images - the one we have in the example